src/main/java/alns/Evaluator.java

```diff
@@ -94,12 +93,85 @@ private static int findSailingDuration(int startTime, Installation fromInstallat
         return ArcGeneration.hourToDiscTimePoint(distance / averageMaxSpeed);
     }
 
+    public static boolean evaluateVisitOrder(Solution solution) {
+        List<List<Order>> orderSequences = solution.getOrderSequences();
+        List<List<Integer>> installationSequences = getInstallationSequences(orderSequences);
+
+        if (installationInMultipleSequences(installationSequences)) {
+            return false;
+        }
+        
+        List<List<Order>> copyOfOrderSequences = Helpers.deepCopyList(orderSequences);
+        if (wrongVisitOrder(copyOfOrderSequences)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static List<List<Integer>> getInstallationSequences(List<List<Order>> orderSequences) {
+        List<List<Integer>> installationSequences = new ArrayList<>();
+
+        for (int vesselNumber = 0; vesselNumber < orderSequences.size(); vesselNumber++) {
+            List<Order> orderSequence = orderSequences.get(vesselNumber);
+            List<Integer> installationSequence = new ArrayList<>();
+            for (Order order : orderSequence) {
+                int installationID = order.getInstallationId();
+                installationSequence.add(installationID);
+            }
+
+            installationSequences.add(installationSequence);
+        }
+
+        return installationSequences;
+    }
+
+    private static boolean installationInMultipleSequences(List<List<Integer>> installationSequences) {
+        while (installationSequences.size() > 1) {
+            for (int i = 1; i < installationSequences.size(); i++) {
+                List<Integer> firstSequence = installationSequences.get(0);
+                List<Integer> secondSequence = installationSequences.get(i);
+                boolean differentInstallation = Collections.disjoint(firstSequence,secondSequence);
+                if(!differentInstallation) {
+                    return true;
+                }
+            }
+
+            installationSequences.remove(0);
+        }
+
+        return false;
+    }
+
+    private static boolean wrongVisitOrder(List<List<Order>> orderSequences) {
+        for (List<Order> orderSequence : orderSequences) {
+            while (orderSequence.size() > 1) {
+                for (int j = 1; j < orderSequence.size(); j++) {
+                    Order firstOrder = orderSequence.get(0);
+                    Order secondOrder = orderSequence.get(j);
+                    if (firstOrder.getInstallationId() == secondOrder.getInstallationId()) {
+                        if (!firstOrder.isDelivery()) {
+                            return true;
+                        }
+                        if (!firstOrder.isMandatory()) {
+                            return true;
+                        }
+                    }
```

Comment:
Tror ikke dette fanger opp alt. Hva om en installasjons ordre står spredt i sekvensen?

Reply:
Har foreslått et oppsett som gjør følgende for alle ordre som ikke er naboen til ordren som undersøkes:

1. Dersom installasjons-ID til ordre i og j er lik og i og j har minst to ordre i mellom, returner true
2. Dersom installasjons-ID til ordre i og j er lik og i og installasjonen mellom de to har ulik installasjons-ID, returner true
3. Dersom installasjons-ID til ordre i og j er lik og ordren i mellom ikke er verken mandatory eller delivery, returner true

Dette oppsettet pushes i neste omgang!